internals/scripts/check-npm-version.js

```diff
@@ -0,0 +1,25 @@
+const chalk = require('chalk');
```

Comment:
If this is used in npm `preinstall`, we really cannot use `chalk` here, as there may be no npm modules installed yet 😉

Reply:
Good point. Same thing for `compare-versions` on line 3. :(

I think I thought of this and just told myself it shouldn't be an issue since we tell users to start with the setup script anyway. (And the setup script has presetup which installs the dependencies for the setup process.)

But yeah this might not be good enough in this situation. If at any point in development a dev needs to delete `node_modules`, they won't be able to `npm install` afterwards.

Is the only solution to avoid usage of `chalk` and `compare-versions`? Because the idea behind using `compare-versions` was that the old logic was somewhat broken. (`parseFloat(stdout)` is not semVer-compatible.)

Reply:
I'm actually not convinced of the value, as a whole, to even checking if the NPM version is there. Why aren't we checking node, and/or other aspects to ensure the user's env is up to spec?

To me, `package.json` denotes minimum reqs in `engines`, and whilst NPM has offered, in the past, `engine-strict` to error, instead of warn, it just seems like another piece the maintainers, of this bp, have to keep up-to-date, for very little upside.

If we decide to keep the check, then we have to accept the minimal check we use, with either `parseFloat` or `parseInt(stdout, 10)`, to check major version reqs, as a means to reduce the number of issues posted due to NPM requirements not being met. We cannot rely on existing NPM packages to be in place, because these packages may not work themselves on earlier/unsupported--by this bp--versions of NPM.

Just my thoughts...

Reply:
Good points. I think I'll push a PR to remove it.

Reply:
(I meant push a commit, not a PR. It's ready.)

Reply:
Nice work on this. 👍